### ISSUE TITLE
[BUG] SteamVR unable to open Room Setup/Tutorial and games

### ISSUE THREAD

**Issue by Kagukara** [ValveSoftware/SteamVR-for-Linux#574]
**Describe the bug**
When I start up SteamVR I get two popups one saying "Room Setup - Please perform Room Setup to begin using SteamVR." and the other "SteamVR - Could not start SteamVR Room Setup: Another application was already starting.".

**To Reproduce**
Steps to reproduce the behaviour:
1. Turn on lighthouses and plugin headset
2. Start SteamVR on Steam

**Expected behaviour**
This is the first time I'm ever trying VR, and I'd expect it to launch SteamVR Room Setup.

**System Information:**
- Distribution: Arch Linux
- SteamVR version: 1.25.7
- Steam client version: Mar 24 2023, at 17:20:43
- Opted into Steam client beta?: No
- Graphics driver version: Mesa 23.1.0-devel (git-4b0b75c27a)
- Gist for SteamVR System Information: https://gist.github.com/Kagukara/96a31958bcfda5c0debf06d01b9c4fc9

**Screenshots**
![capture_2023-04-02_13-09-41](https://user-images.githubusercontent.com/43391109/229352036-36fda846-fc9f-43e9-8c37-541762f5ac7b.png)
![capture_2023-04-02_13-09-43](https://user-images.githubusercontent.com/43391109/229352040-64c1feed-1bff-40f3-a6da-5841cde86434.png)
![capture_2023-04-02_13-21-13](https://user-images.githubusercontent.com/43391109/229352484-2a052ec7-5007-409b-ba88-62777f9dab65.png)

**Additional context**
Closing both the popups and clicking either Room Setup or Tutorial in the burger menu gives the "SteamVR - could not start SteamVR Room Setup/Tutorial: Another application was already starting." popup.

**More System Information:**
- WM: sway (I have tried in i3, and I get the same result)
- CPU: AMD Ryzen 9 5950X 16-Core
- GPU: AMD Radeon RX 7900 XTX (gfx1100, LLVM 16.0.0, DRM 3.49, 6.2.9-arch1-1)
- PipeWire: Compiled with libpipewire 0.3.67, Linked with libpipewire 0.3.67
Not sure if this additional information would be helpful.

**Comment by Kagukara** [comment]
When I have SteamVR open, I can launch SteamVR Room Setup via the terminal with:
`~/.local/share/Steam/steamapps/common/SteamVR/tools/steamvr_room_setup/linux64/steamvr_room_setup`

Which outputs:
```
Found path: /home/richard/.local/share/Steam/steamapps/common/SteamVR/tools/steamvr_room_setup/linux64/steamvr_room_setup
Mono path[0] = '/home/richard/.local/share/Steam/steamapps/common/SteamVR/tools/steamvr_room_setup/linux64/steamvr_room_setup_Data/Managed'
Mono path[1] = '/home/richard/.local/share/Steam/steamapps/common/SteamVR/tools/steamvr_room_setup/linux64/steamvr_room_setup_Data/Mono'
Mono config path = '/home/richard/.local/share/Steam/steamapps/common/SteamVR/tools/steamvr_room_setup/linux64/steamvr_room_setup_Data/Mono/etc'
Preloaded 'ScreenSelector.so'
Preloaded 'libopenvr_api.so'
```
This loads up Room Setup, but when I go to either "ROOM-SCALE" or "STANDING ONLY" the headset is shown as off (and the controllers). In SteamVR however it is on.
 
![capture_2023-04-02_15-51-46](https://user-images.githubusercontent.com/43391109/229361318-abff02c2-8c4f-470a-b8a1-dfa7327a15a7.png)
![capture_2023-04-02_15-52-35](https://user-images.githubusercontent.com/43391109/229361323-2df287bb-1e31-4d97-9266-ddec8062c692.png)
![capture_2023-04-02_15-52-50](https://user-images.githubusercontent.com/43391109/229361326-e711c800-0f93-4fd1-849b-d8831ee54abd.png)
_Controllers are off in this picture as I was testing "STANDING ONLY"_

**Comment by Kagukara** [comment]
With SteamVR open and running Room Setup from the terminal, I get the same as above. For Tutorial I get this:
`~/.local/share/Steam/steamapps/common/SteamVR/tools/steamvr_tutorial/linux64/steamvr_tutorial`
```
Found path: /home/richard/.local/share/Steam/steamapps/common/SteamVR/tools/steamvr_tutorial/linux64/steamvr_tutorial
Mono path[0] = '/home/richard/.local/share/Steam/steamapps/common/SteamVR/tools/steamvr_tutorial/linux64/steamvr_tutorial_Data/Managed'
Mono path[1] = '/home/richard/.local/share/Steam/steamapps/common/SteamVR/tools/steamvr_tutorial/linux64/steamvr_tutorial_Data/Mono'
Mono config path = '/home/richard/.local/share/Steam/steamapps/common/SteamVR/tools/steamvr_tutorial/linux64/steamvr_tutorial_Data/Mono/etc'
Preloaded 'ScreenSelector.so'
Preloaded 'libaudioplugin_phonon.so'
Preloaded 'libopenvr_api.so'
Preloaded 'libphonon.so'
Invalid serialized file version. File: "/home/richard/.local/share/Steam/steamapps/common/SteamVR/tools/steamvr_tutorial/linux64/steamvr_tutorial_Data/globalgamemanagers". Expected version: 5.6.0b7. Actual version: 2017.3.1f1.
 
(Filename:  Line: 1244)

PlayerInitEngineNoGraphics settings: Failed to load PlayerSettings (internal index #0).
Most likely data file is corrupted, or built with mismatching
editor and platform support versions.
Failed to initialize player
```

**Comment by Kagukara** [comment]
I have a very janky workaround for playing VR games. I decided to try VR in Minecraft Java Edition (downloaded from the AUR) using Fabric and Vivecraft mod. At first, it didn't work because I had run SteamVR before loading the game, so I closed everything down and just launched Minecraft. My Index was then able to display Minecraft in VR. I then went to look and see if SteamVR was running, and it was. I closed Minecraft down, but SteamVR kept running, so I launched a VR game. It didn't work. So I restarted the process, and after closing Minecraft, I put -hmd=SteamVR in the launch options for the game I wanted to play, and it worked. But when switching to a new game, I have to kill the SteamVR process in the task manager, as doing it through Steam doesn't work, and repeat the process (opening Minecraft in VR, etc.).

I have no idea why this works, and launching SteamVR and running a game from within SteamVR doesn't, but it's a step forward.

Launching SteamVR still has the same problems. When doing the workaround, you cannot open the SteamVR menu in VR or see the SteamVR interface on the PC.

**Comment by goarvic** [comment]
I think I am affected by the same problem. For several days SteamVRHome does not start

**Comment by Kagukara** [comment]
Here is a comparison between the process launched between SteamVR launched through steam and running Minecraft with the Vivecraft mod.

Launched via Steam:
![Steam-SteamVR](https://user-images.githubusercontent.com/43391109/230352328-be69d750-765d-41f6-a6f9-efeba9b38a58.png)

Launched via Minecraft with the Vivecraft mod - Minecraft Open:
![minecraft-vivecraft-open](https://user-images.githubusercontent.com/43391109/230352790-1307da0f-1513-4c0d-b841-372ab7ff87be.png)

Launched via Minecraft with the Vivecraft mod - Minecraft Closed:
![minecraft-vivecraft-closed](https://user-images.githubusercontent.com/43391109/230352905-d80adcd9-9766-4608-827e-d56ee0033465.png)

**Comment by hparadiz** [comment]
Same problem on Gentoo

**Comment by Kagukara** [comment]
Here is an updated version of running steam via the terminal and launching steamvr: https://pastebin.com/D8R2ZNT5 (11:28 ~ 14/04/2023)

Still have the same problem with the workaround being the only way to launch games.

**Comment by Kagukara** [comment]
From a reply I got from a post I made on the [SteamVR Linux Discussion](https://steamcommunity.com/app/250820/discussions/5/3827539207872101653/) I have now been able to click on both Tutorial and Room Setup without the popup "Could not start SteamVR Room Setup(Tutorial): Another application was already starting." appearing.

This has allowed me to do Room Setup and launch games in SteamVR, without having to use the vivecraft workaround.

The issue still persists with SteamVR Home on, with the vivecraft mod workaround being the only way I've found to launch games.

**Comment by Kagukara** [comment]
@kisak-valve any chance you could take a look at this, the problem still persists with everything up to date.

It looks like steamtours.sh could be the main culprit from looking through my previous logs.

**Comment by kisak-valve** [comment]
Hello @Kagukara, friendly reminder that I'm a moderator for Valve's issue trackers on Github, and not a SteamVR developer myself. I don't have any special insight into this issue.

Personally, I don't autostart SteamVR Home on my test box, primarily for historical consistency with some video driver tests and that coincidentally avoided this issue.

**Comment by Kagukara** [comment]
Thank you for the reply, just kept seeing you reply to other issues and thought you maybe able to help.

**Comment by PistolRcks** [comment]
> I have a very janky workaround for playing VR games. I decided to try VR in Minecraft Java Edition (downloaded from the AUR) using Fabric and Vivecraft mod. At first, it didn't work because I had run SteamVR before loading the game, so I closed everything down and just launched Minecraft. My Index was then able to display Minecraft in VR. I then went to look and see if SteamVR was running, and it was. I closed Minecraft down, but SteamVR kept running, so I launched a VR game. It didn't work. So I restarted the process, and after closing Minecraft, I put `-hmd=SteamVR` in the launch options for the game I wanted to play, and it worked. But when switching to a new game, I have to kill the SteamVR process in the task manager, as doing it through Steam doesn't work, and repeat the process (opening Minecraft in VR, etc.).
> 
> I have no idea why this works, and launching SteamVR and running a game from within SteamVR doesn't, but it's a step forward.
> 
> Launching SteamVR still has the same problems. When doing the workaround, you cannot open the SteamVR menu in VR or see the SteamVR interface on the PC.

Oddly, this fix works for me as well. One of the supporting applications (vrwebhelper?) is disconnecting the entire thing from Steam (which will not allow any apps to launch), or so I can tell (or perhaps this is intended functionality?).

Relevant log information from my personal `vrserver.txt`
```
Sat Jun 17 2023 15:26:25.599889 - Unable to read message from socket: 0
Sat Jun 17 2023 15:26:25.599909 - Process steam (40437) disconnected (Thread(0x0x7f77e4016750/0x0x7f)
Sat Jun 17 2023 15:26:25.599930 - AppInfoManager.ProcessQuit processid=40437 eLaunchingApp=LaunchingApp_None
Sat Jun 17 2023 15:26:25.599946 - AppInfoManager.ProcessQuit: Clearing application system.generated.steam PID was 40437
Sat Jun 17 2023 15:26:25.599961 - AppInfoManager.ProcessQuit: Clearing application system.generated.steam PID because 40437 has exited
```

**Comment by Nebulon-BFrigateFTW** [comment]
Had this problem as well on OpenSUSE Tumbleweed. Interestingly, even after disabling SteamVR Home, I also get some error code 109 whenever I start SteamVR, but this doesn't seem to actually cause problems.
Also, could this perhaps be identical to #579 ?

**Comment by PistolRcks** [comment]
Seems to be resolved as of beta update 1.26.5? That update also seems to have messed with the colors of the dashboard, but that's outside of the scope of this issue.

**Comment by Kagukara** [comment]
Just tried SteamVR Beta 1.26.5, turned on SteamVR Home in the dashboard and tried to launch SteamVR Room Setup and a game. 

Launching SteamVR Room Setup gives the same response as it did at the start of this issue:
![capture_2023-07-05_14-44-25](https://github.com/ValveSoftware/SteamVR-for-Linux/assets/43391109/3dfeb587-f126-49fa-872e-9835310037be)

Launching a game doesn't give any response and just leaves the game on launching but never launches.

The colours in the beta are also messed up but I think there is an issue on this already.
Beta:
![capture_2023-07-05_14-48-26](https://github.com/ValveSoftware/SteamVR-for-Linux/assets/43391109/3fe33f85-cc3d-4718-98cf-8a93aa096523)

Non-Beta:
![capture_2023-07-05_14-52-02](https://github.com/ValveSoftware/SteamVR-for-Linux/assets/43391109/74b85bb5-63e9-400f-adb1-e3394b5f0506)

**Comment by eyeonus** [comment]
I'm having this problem as well, but I can't even open the options in order to turn off Home, so I don't know if that would fix it for me. Is there anyway to disable it that doesn't require starting VR?

**Comment by Kagukara** [comment]
> Is there anyway to disable it that doesn't require starting VR?

You should be able to open SteamVR, click the burger menu icon, settings, general, and the SteamVR Home option should be at the bottom.
![capture_2023-07-09_11-42-15](https://github.com/ValveSoftware/SteamVR-for-Linux/assets/43391109/c4fa8b9e-b208-4ab8-955f-36bc4ab47b94)

**Comment by eyeonus** [comment]
> > Is there anyway to disable it that doesn't require starting VR?
> 
> You will need to start SteamVR with your headset on but the other way would be to open SteamVR, click the burger menu icon, settings, general, and the SteamVR Home option should be at the bottom. ![capture_2023-07-09_11-42-15](https://user-images.githubusercontent.com/43391109/252140318-c4fa8b9e-b208-4ab8-955f-36bc4ab47b94.png)

I can not. nothing happens when I click the burger. This is why I asked if there was a way to do it that DOESN'T require launching VR.

**Comment by Kagukara** [comment]
> This is why I asked if there was a way to do it that DOESN'T require launching VR.

Ah, my misunderstanding. I went to thinking "in VR" and not "launching/starting VR (SteamVR)". I'm not sure if you can change the setting in one of the files in the SteamVR common files. So you may want to create your own issue for not being able to click the burger icon if you're not able to click it.

**Comment by Kagukara** [comment]
@kisak-valve This has been retested: [link to comment](https://github.com/ValveSoftware/SteamVR-for-Linux/issues/574#issuecomment-1621809379)

**Comment by WhitleyStriber** [comment]
Same problem here, except I don't get any message on my desktop, and my headset says Launching Room Setup.

**Comment by susrn** [comment]
I was having the same issue until I ran `steamvr_room_setup.sh` and noticed that it was trying to access `vrenv.sh` in the wrong directory.

One of the lines of the output I have is like this:
`~/.local/share/Steam/steamapps/common/SteamVR/tools/steamvr_room_setup/linux64/../../../runtime/bin/vrenv.sh: No such file or directory`.
Apparently the `runtime/` directory doesn't exist, so I looked at `steamvr_room_setup.sh` on line 39, and removed `runtime/` from `VRBINDIR=${ROOT}/../../../runtime/bin`.

After doing that I've restarted SteamVR and that seems to have fixed it!

**Comment by Hicks79** [comment]
> I was having the same issue until I ran `~/.local/share/Steam/steamapps/common/SteamVR/tools/steamvr_room_setup/linux64/steamvr_room_setup.sh` and noticed that it was trying to access `vrenv.sh` in the wrong directory.
> 
> One of the lines of the output I have is like this: `~/.local/share/Steam/steamapps/common/SteamVR/tools/steamvr_room_setup/linux64/../../../runtime/bin/vrenv.sh: No such file or directory`. Apparently the `runtime/` directory doesn't exist, so I looked at `steamvr_room_setup.sh` on line 39, and removed `runtime/` from `VRBINDIR=${ROOT}/../../../runtime/bin`.
> 
> After doing that I've restarted SteamVR and that seems to have fixed it!

THANKS @susrn 

How did you know the route was incorrect?
   #VRBINDIR=${ROOT}/../../../runtime/bin
    VRBINDIR=${ROOT}/../../../bin

the ROOT-variable had this:
ROOT="$(cd "$(dirname "$0")" && echo $PWD)"

I can at least see the room-setup now, even though there are other bugs.

Thanks my friend! :D

Why doesn't valve fix these things? I hope they are with HL3.

**Comment by TTimo** [comment]
This should be fixed in latest beta. Please open a new issue if you believe that is not the case. Make sure you do not have any locally modified SteamVR scripts left over.

**Comment by atomicptr** [comment]
I'm on beta and also had to modify the script to be able to do room setup (although the line moved down somewhat)

**Comment by trashhead101** [comment]
To sum it up, 
1. go into the Folder `~/.local/share/Steam/steamapps/common/SteamVR/tools/steamvr_room_setup/` (basically: "/home/name/.local/share/Steam/steamapps/common/SteamVR/tools/steamvr_room_setup/linux64/")
2. open steamvr_room_setup.sh
and change line 51 `VRBINDIR=${ROOT}/../../../runtime/bin` to `VRBINDIR=${ROOT}/../../../bin` (remove the 'runtime/')

(3. only for me (Plasma, Arch Linux): make sure you got openssl (for Qt) installed `sudo pacman -S openssl`)

(Also, I installed `lib32-libvdpau`, `lib32-gtk2`, `lib32-libva` a few seconds before, which you generally need for steamVR (I guess), so make sure you got them also)

**Comment by KvaGram** [comment]
I had a simular problem when trying to launch room setup.
I have just moved, so I needed to set it up again.
When I tried, the application seemingly did not launch.

Then I tried launching from terminal as you suggested .
There the window opened briefly, then closed, reporting a segment fault in the terminal.

I then attempted to switch to "Linux_1.14". The VR had to redo the root access setup, and complained about something crashing. Yet SteamVR did launch. Then I once again tried to launch Room Setup, and it worked perfectly.
After switching back to the beta-branch, SteamVR and the home enviroment loaded succesfully, with the new room setup all done, and ready to use.
